dynatrace/src/main/java/com/dynatrace/opentelemetry/metric/OneAgentMetadataEnricher.java

```diff
@@ -70,57 +70,110 @@ public OneAgentMetadataEnricher(Logger logger) {
 
       // make sure key and value are set to non-null, non-empty values
       if ((key == null || key.isEmpty()) || (value == null || value.isEmpty())) {
-        logger.warning(String.format("could not parse OneAgent metadata line ('%s')", line));
+        getLogger().warning(String.format("could not parse OneAgent metadata line ('%s')", line));
         continue;
       }
       entries.add(new AbstractMap.SimpleEntry<>(key, value));
     }
     return entries;
   }
 
-  private List<String> getMetadataFileContent() {
-    String indirectionBaseName = "dt_metadata_e617c525669e072eebe3d0f08212e8f2";
-    String secretFileName = null;
-    try (BufferedReader reader =
-        new BufferedReader(new FileReader(String.format("%s.properties", indirectionBaseName)))) {
-      String line = "";
-      // this whole while block is to make sure this function will still work even if the contents
-      // of the
-      // indirection file were to change in the future.
+  /**
+   * Get the file name of the file to which OneAgent persists its parameters.
+   *
+   * @param fileContents A {@link Reader} object pointing at the indirection file. Will be read
+   *     using a {@link BufferedReader}
+   * @param indirectionBaseName The prefix for the OneAgent metadata file. Lines in the indirection
+   *     file without this prefix are ignored. If null is passed as this parameter,
+   *     'dt_metadata_e617c525669e072eebe3d0f08212e8f2' is used.
+   * @return The string containing the filename, with no leading or trailing whitespace.
+   * @throws IOException if an error occurs during reading of the file.
+   */
+  String getIndirectionFilename(Reader fileContents, String indirectionBaseName)
+      throws IOException {
+    if (fileContents == null) {
+      throw new IOException("passed Reader cannot be null.");
+    }
+    String prefix = indirectionBaseName;
+    if (indirectionBaseName == null) {
+      prefix = "dt_metadata_e617c525669e072eebe3d0f08212e8f2";
+    }
 
+    String oneAgentMetadataFileName = null;
+    String line;
+
+    try (BufferedReader reader = new BufferedReader(fileContents)) {
       // read file line by line, and stop if the end of the file is reached.
       while ((line = reader.readLine()) != null) {
+        line = line.trim();
         // the secret file contains the basename and a random number at the end.
-        if (line.contains(indirectionBaseName)) {
-          secretFileName = line;
+        if (line.startsWith(prefix)) {
```

Comment:
I don't think you can rely on the actual file name to start with a particular prefix, that's not defined in the OneAgent API and therefore subject to change.